src/operator/catch.ts

```diff
@@ -64,16 +64,16 @@ import { subscribeToResult } from '../util/subscribeToResult';
  * @name catch
  * @owner Observable
  */
-export function _catch<T, R>(this: Observable<T>, selector: (err: any, caught: Observable<T>) => ObservableInput<R>): Observable<T | R> {
+export function _catch<T, R = T>(this: Observable<T>, selector: (err: any, caught: Observable<R>) => ObservableInput<R>): Observable<R> {
   const operator = new CatchOperator(selector);
   const caught = this.lift(operator);
   return (operator.caught = caught);
 }
 
-class CatchOperator<T, R> implements Operator<T, T | R> {
-  caught: Observable<T>;
+class CatchOperator<T, R = T> implements Operator<T, R> {
+  caught: Observable<R>;
```

Comment:
This was failing without this change, I wasn't entirely sure it made sense, but looking at the code it should have always been `Observable<T | R>` and never `Observable<T>`